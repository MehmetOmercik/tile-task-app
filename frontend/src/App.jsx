import { useEffect, useState } from "react";
import { Tile } from "./components/tile";
import { TaskSlider } from "./components/taskSlider";
import { getTiles } from "./utils/http";
import { Overlay } from "./components/overlay";

export default function App() {
  const [tiles, setTiles] = useState([]);
  const [tileStatus, setTileStatus] = useState("live");
  const [tasks, setTasks] = useState([]);
  const [openOverlay, setOpenOverlay] = useState(false);

  useEffect(() => {
    const handleMount = async () => {
      const response = await getTiles(tileStatus);
      setTiles(response);
    };
    handleMount();
  }, [tileStatus]);

  const handleOverlay = async () => {
    setOpenOverlay(!openOverlay);
  };
  const filterButtonStyle =
    "border p-4 border-red-700 bg-blue-400 rounded-2xl hover:bg-blue-200";

  return (
    <>
      <div className="bg-blue-500 p-5 my-10 flex gap-x-10 justify-center items-center text-3xl">
        <p className="mr-5">Status Filter: </p>
        <button
          onClick={() => {
            setTileStatus("live");
          }}
          className={filterButtonStyle}
        >
          Live
        </button>
        <button
          onClick={() => {
            setTileStatus("pending");
          }}
          className={filterButtonStyle}
        >
          Pending
        </button>
        <button
          onClick={() => {
            setTileStatus("archived");
          }}
          className={filterButtonStyle}
        >
          Archived
        </button>
      </div>
      <div className="flex items-start justify-center gap-x-10 gap-y-4 m-4 flex-wrap">
        {tiles.map((tile) => {
          return (
            <Tile
              key={tile.id}
              id={tile.id}
              launchDate={tile.launch_date}
              status={tile.status}
              handleOverlay={handleOverlay}
              setTasks={setTasks}
            />
          );
        })}
        <Overlay isOpen={openOverlay} onClose={handleOverlay}>
          <TaskSlider tasks={tasks} />
        </Overlay>
      </div>
    </>
  );
}
