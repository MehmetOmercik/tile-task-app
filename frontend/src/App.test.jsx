import App from "./App";
import { renderWithProviders } from "./test/test-utils";

describe("app tests", () => {
  test("snapshot test", () => {
    const app = renderWithProviders(<App />);
    expect(app).toMatchSnapshot();
  });
});
