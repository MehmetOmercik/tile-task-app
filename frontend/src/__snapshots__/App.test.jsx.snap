// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`app tests > snapshot test 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="absolute bg-gray-400 right-5 mt-3 text-7xl border px-2 rounded-xl hover:bg-gray-200"
      >
        +
      </button>
      <div
        class="bg-blue-500 p-5 my-10 flex gap-x-10 justify-center items-center text-3xl"
      >
        <p
          class="mr-5"
        >
          Status Filter: 
        </p>
        <button
          class="border p-4 border-red-700 bg-blue-400 rounded-2xl hover:bg-blue-200"
        >
          Live
        </button>
        <button
          class="border p-4 border-red-700 bg-blue-400 rounded-2xl hover:bg-blue-200"
        >
          Pending
        </button>
        <button
          class="border p-4 border-red-700 bg-blue-400 rounded-2xl hover:bg-blue-200"
        >
          Archived
        </button>
      </div>
      <div
        class="flex items-start justify-center gap-x-10 gap-y-4 m-4 flex-wrap"
      />
    </div>
  </body>,
  "container": <div>
    <button
      class="absolute bg-gray-400 right-5 mt-3 text-7xl border px-2 rounded-xl hover:bg-gray-200"
    >
      +
    </button>
    <div
      class="bg-blue-500 p-5 my-10 flex gap-x-10 justify-center items-center text-3xl"
    >
      <p
        class="mr-5"
      >
        Status Filter: 
      </p>
      <button
        class="border p-4 border-red-700 bg-blue-400 rounded-2xl hover:bg-blue-200"
      >
        Live
      </button>
      <button
        class="border p-4 border-red-700 bg-blue-400 rounded-2xl hover:bg-blue-200"
      >
        Pending
      </button>
      <button
        class="border p-4 border-red-700 bg-blue-400 rounded-2xl hover:bg-blue-200"
      >
        Archived
      </button>
    </div>
    <div
      class="flex items-start justify-center gap-x-10 gap-y-4 m-4 flex-wrap"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
